Lib/unittest/mock.py

```diff
@@ -2816,6 +2819,59 @@ def __set__(self, obj, val):
         self(val)
 
 
+class ThreadingMock(MagicMock):
+    """
+    A mock that can be used to wait until on calls happening
+    in a different thread.
+    """
+
+    def __init__(self, *args, **kwargs):
+        _safe_super(ThreadingMock, self).__init__(*args, **kwargs)
+        self._event = threading.Event()
```

Comment:
Are we going to allow customization of the event class?

Reply:
From a chat with @voidspace (see https://github.com/python/cpython/pull/16094#issuecomment-531314327), passing `multiprocessing.Event` does not work. It needs further support especially if we want to have the `wait_until_any_call`. That is also why we renamed the mock to be specific to threading. Once we get this through we can look at supporting multiprocessing if there is a need.